Documentation/ApiOverview/Database/ExpressionBuilder/Index.rst

```diff
@@ -184,25 +184,265 @@ a query builder with the connection pool.
 
+..  _database-expression-builder-repeat:
+
+:php:`ExpressionBuilder::repeat()`
+----------------------------------
+
+..  versionadded:: 13.1
+
+..  include:: _ExpressionBuilderRepeat.rst.txt
+
+Create a statement to generate a value repeating defined :php:`$value` for
+:php:`$numberOfRepeats` times. This method can be used to provide the
+repeat number as a sub-expression or calculation.
+
+:sql:`REPEAT(string, number)` is used to build this expression for all database
+vendors except SQLite for which the compatible replacement construct expression
+:sql:`REPLACE(PRINTF('%.' || <valueOrStatement> || 'c', '/'),'/', <repeatValue>)`
+is used, based on :sql:`REPLACE()` and the built-in :sql:`printf()`.
+
+..  warning::
+
+    Be aware to properly quote values, identifiers and sub-expressions.
+    No automatic quoting will be applied.
+
+..  literalinclude:: _RepositoryRepeat.php
+    :language: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
+
+..  _database-expression-builder-right:
+
+:php:`ExpressionBuilder::right()`
+----------------------------------
+
+..  versionadded:: 13.1
+
+..  include:: _ExpressionBuilderRight.rst.txt
+
+Extract :php:`$length` character of :php:`$value` from the right side.
+
+Creates a :sql:`RIGHT(string, number_of_chars)` expression for all supported
+database vendors except SQLite, where :sql:`substring(string, integer[, integer])`
+is used to provide a compatible expression.
+
+..  warning::
+
+    Be aware to properly quote values, identifiers and sub-expressions.
+    No automatic quoting will be applied.
+
+..  literalinclude:: _RepositoryRight.php
+    :language: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
+
+
+..  _database-expression-builder-rightPad:
+
+:php:`ExpressionBuilder::rightPad()`
+----------------------------------
+
+..  versionadded:: 13.1
+
+..  include:: _ExpressionBuilderRightPad.rst.txt
+
+Right-pad the value or sub-expression result with :php:`$paddingValue`, to a
+total length of :php:`$length`.
+
+SQLite does not support :sql:`RPAD(string, integer, string)`, therefore a
+complexer compatible replacement expression construct is created.
+
+..  warning::
+
+    Be aware to properly quote values, identifiers and sub-expressions.
+    No automatic quoting will be applied.
+
+..  literalinclude:: _RepositoryRightPad.php
+    :language: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyTableRepository.php
+
+..  _database-expression-builder-space:
+
+:php:`ExpressionBuilder::space()`
+----------------------------------
+
+..  versionadded:: 13.1
+
+..  include:: _ExpressionBuilderSpace.rst.txt
+
+Create statement containing :php:`$numberOfSpaces` spaces.
+
+The :sql:`SPACE(number)` expression is used for MariaDB and MySQL and
+:php:`ExpressionBuilder::repeat()` expression as fallback for PostgreSQL
+and SQLite.
+
+..  warning::
+
+    Be aware to properly quote values, identifiers and sub-expressions.
```

Comment:
```suggestion
    Be aware to properly quote values, identifiers and sub-expressions.
```

Quotenote. I am thinking, we should maybe outsource this as an include file, so it can centrally be changed for all methods here that repeat this?